returnn/tf/layers/signal_processing.py

```diff
@@ -426,32 +459,46 @@ def _get_nr_of_channels_from_input_placeholder(self):
 
   def _apply_stft_to_input(self):
     from returnn.tf.util.basic import get_shape
-    def _cropStftOutputToReferenceFrameSizeLength(channel_concatenated_stft, crop_size):
-      return tf.slice(channel_concatenated_stft, [0, 0, 0], [get_shape(channel_concatenated_stft)[0], crop_size, get_shape(channel_concatenated_stft)[2]])
+
+    def _crop_stft_output_to_reference_frame_size_length(channel_concatenated_stft_, crop_size):
+      return tf.slice(channel_concatenated_stft_, [0, 0, 0], [get_shape(channel_concatenated_stft_)[0], crop_size,
+                                                              get_shape(channel_concatenated_stft_)[2]])
 
     input_placeholder = self.input_data.get_placeholder_as_batch_major()
     channel_wise_stft_res_list = list()
     for fft_size, frame_size in zip(self._fft_sizes, self._frame_sizes):
       def _get_window(window_length, dtype):
         if self._window == "hanning":
-            window = tf.contrib.signal.hann_window(window_length, dtype=dtype)
-        if self._window == "blackman":
-            tf_compat.v1.assert_equal(frame_size, window_length)
-            import scipy.signal
-            window = tf.constant(scipy.signal.blackman(frame_size), dtype=tf.float32)
-        if self._window == "None" or self._window == "ones":
+          try:
+            from tensorflow.signal import hann_window
+          except ImportError:
+            from tensorflow.contrib.signal import hann_window
+          window = hann_window(window_length, dtype=dtype)
+        elif self._window == "blackman":
+          tf_compat.v1.assert_equal(frame_size, window_length)
```

Comment:
This assert does not have any effect.
You should use `with tf.control_dependencies([...])` instead.

Reply:
What exactly do you mean here? I didn't change anything in this line but can still update it of course.

Reply:
Yea, I know, it was wrong before. But I thought, when you already change this, you should maybe also fix such obvious bugs. Or just remove the assert if you want to have the old behavior.

Reply:
I removed it now.